Sources/SwiftDocC/Model/Rendering/Tutorial/References/DownloadReference.swift

```diff
@@ -60,15 +62,41 @@ public struct DownloadReference: RenderReference, URLReference, Equatable {
     public init(identifier: RenderReferenceIdentifier, renderURL url: URL, sha512Checksum: String) {
         self.init(identifier: identifier, renderURL: url, checksum: sha512Checksum)
     }
-    
+
+    enum CodingKeys: CodingKey {
+        case type
+        case identifier
+        case url
+        case checksum
+    }
+
+    public init(from decoder: Decoder) throws {
+        let container = try decoder.container(keyedBy: CodingKeys.self)
+        self.type = try container.decode(RenderReferenceType.self, forKey: .type)
+        self.identifier = try container.decode(RenderReferenceIdentifier.self, forKey: .identifier)
+        self.url = try container.decode(URL.self, forKey: .url)
+        self.urlWasDecoded = true
```

Comment:
Similar to the other reference, I'm wondering if we need this. It seems like we always encode a URL, so I don't see how we would ever fail to decode one.

Reply:
The `urlWasDecoded` property doesn't denote that it was successfully decoded, it's a note to the `encode(to:)` function to not pass the result through `renderURL(for:)`. If i didn't include this property, the existing round-trip tests failed because the URL was being rewritten by `DownloadReference` before being reloaded by `ExternalLocationReference`.

Reply:
Ohhh got it. Okay makes sense. I think a doc comment on the `urlWasDecoded` might help me when I come across this in the future then but not a blocker.